lib/modules/dosomething/dosomething_global/dosomething_global.module

```diff
@@ -256,3 +273,143 @@ function dosomething_global_get_language($account, stdClass $node = NULL) {
 
   return $language;
 }
+
+/**
+ * Detect from the original request path whether the current request is for
+ * translated (non-Global English) content.
+ *
+ * This assumes that Global English is mapped to a blank ('') URL prefix.
+ *
+ * @return bool
+ */
+function dosomething_global_is_translation_request() {
+
+  $languages = language_list();
+  $matches = array();
+
+  if (preg_match('/([^\/]+)\//', request_path(), $matches)) {
+    foreach ($languages as $sys_lang) {
+
+      // If the prefix matches a language prefix besides that of Global
+      // English:
+      if ($matches[1] == $sys_lang->prefix && $sys_lang->language != 'en-global') {
```

Comment:
Wr should check if `$matches[1]` is set (better `!empty()`) to avoid PHP notices.